src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -642,7 +644,25 @@ private uint getMethodAttribsInternal(MethodDesc method)
                 // RyuJIT special cases this method; it would assert if it's not final
                 // and we might not have set the bit in the code above.
                 result |= CorInfoFlag.CORINFO_FLG_FINAL;
-           }
+            }
+
+            // Check for hardware intrinsics
+            if (owningType.IsIntrinsic && owningType is MetadataType mdType)
```

Comment:
Should similar logic be done for `Vector64<T>`, `Vector128<T>`, and `Vector256<T>` (since they are also currently marked `[Intrinsic]` due to their special ABI requirements)?

Reply:
You mean to report `CORINFO_FLG_JIT_INTRINSIC` on the methods on Vector64/128/256? Can you point me to where that happens? I pretty much copied this part so far:

https://github.com/dotnet/coreclr/blob/80720a220c340e99aa88d04ebae8118eb4fc198b/src/vm/methodtablebuilder.cpp#L1499-L1521

Reply:
> (since they are also currently marked [Intrinsic] due to their special ABI requirements)?

No, `[Intrinsic]` on `Vector64/128/256<T>` is only for "faster" detection.

Reply:
I was referring to us setting `fIsIntrinsicType` here: https://github.com/dotnet/coreclr/blob/master/src/vm/methodtablebuilder.cpp#L1523-L1538

For `Vector64<T>`, `Vector128<T>`, and `Vector256<T>` we then check for that flag and update the packing and size of the type here: https://github.com/dotnet/coreclr/blob/master/src/vm/methodtablebuilder.cpp#L9518-L9590
* This ensures that layout of the struct is correct when interoping with native code and is always applicable, even when the individual ISAs aren't supported or enabled
* We also have individual methods in these type marked as `[Intrinsic]` which is handled here: https://github.com/dotnet/coreclr/blob/master/src/vm/methodtablebuilder.cpp#L5123 (but these have software fallbacks)

For `Vector<T>`, we then specially handle the size here: https://github.com/dotnet/coreclr/blob/master/src/vm/methodtablebuilder.cpp#L1153-L1215
* This isn't really applicable to CoreRT, unless you support specifiying AVX2 as the "target" ISA

Reply:
Ah yes, we do those. Vector64/128/256 should be covered by #6833 and `Vector<T>` by #3678.

Reply:
Wow, CoreRT starts to support vectors! 
Personally, I am really interested in CoreRT project as well. Is there anything I can help? If yes, where can I start to look into? 😄 (not limited to SIMD or intrinsic)

Reply:
> What typically works best is that you pick a scenario or app that you are interested in and you make that work.

Thanks for the reply, this is a good idea to ramp up. The IL interpreter looks really interesting, I will start to take a look.

Reply:
#6182 has some details on how to get started with the interpreter. Help will definitely be appreciated! There's a fundamental change to how IL is parsed in the interpreter in flight: https://github.com/dotnet/corert/pull/6725#discussion_r245397687